### ISSUE TITLE
DeMixT

### ISSUE THREAD

**Issue by fgao-github** [Bioconductor/Contributions#853]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/wwylab/DeMixT

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by mtmorgan** [comment]
The technical issues on our end have been worked through. However, your package does not have a vignette and does not [re-use standard Bioconductor representations][1] such as SummarizedExperiment, so is not appropriate for Bioconductor. Will you update your package so that it can be reviewed?

[1]: http://bioconductor.org/developers/how-to/commonMethodsAndClasses/

**Comment by mtmorgan** [comment]
Will you be updating your package to allow it to be added to the review queue, or should we close the issue due to inactivity?

**Comment by fgao-github** [comment]
We will update it next week. Thank you for your patience. 

Thanks,
Fan

**Comment by mtmorgan** [comment]
I'm closing this issue; please post a comment including @mtmorgan if you decide to continue your package development.

**Comment by fgao-github** [comment]
Hi Biconductor team,

Could you please reopen the issue for us?

I will focus on the DeMixT package and submit the revised package  
within this week.

Thanks very much,
Fan




Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
It looks like the package still does not have a vignette, etc. Please update it to Bioconductor submission standards and I will re-open the issue then.

**Comment by fgao-github** [comment]
Hi Martin,

Sure. I will follow your standards, and will let you know immediately  
when I finish it.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by fgao-github** [comment]
Hi Martin,

Attached is our vignette for DeMixT. Do we need to upload it on GitHub  
(https://github.com/wwylab/DeMixT)? Could you please help re-open the  
issue for us?

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by fgao-github** [comment]
Hello,

I just upload the vignette folder to Github  
(https://github.com/wwylab/DeMixT). Could you please reopen the issue  
for us and let us know the next steps?

Thanks,
Fan


Quoting lshep <notifications@github.com>:

**Comment by mtmorgan** [comment]
The vignette needs to be in a folder `vignettes/`, see https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Writing-package-vignettes .

How have you addressed the concerns about re-using standard R data structures ?

**Comment by fgao-github** [comment]
Hi Martin,

We have addressed both the standard R data (SummarizedExperiment)  
issue. We also put the RMD file into vignettes/ folder.

Below is our GitHub repository:
https://github.com/wwylab/DeMixT

Please let us know if there are any questions.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
Hi -- there is no vignettes/ directory in your github repository. Here's the directory structure of a simple package that I am actively working on

```
RedisParam master$ tree
.
├── DESCRIPTION
├── NAMESPACE
├── R
│   └── RedisParam-class.R
├── man
│   └── RedisParam.Rd
├── tests
│   ├── testthat
│   │   └── test_RedisParam-class.R
│   └── testthat.R
└── vignettes
    └── RedisParam-intro.Rmd
```
The file 'RedisParam-intro.Rmd' contains the vignette for the package; the vignette is in a directory called 'vignettes' in the top level of the package repository.

**Comment by fgao-github** [comment]
Hi Martin,

I added the vignettes files to GitHub. Below is our GitHub repository.
https://github.com/wwylab/DeMixT

Please let us know if there are any questions.

Thanks,
Fan

Quoting Martin Morgan <notifications@github.com>:

**Comment by fgao-github** [comment]
Hi Martin,

Have you made any progress about our DeMixT package? (https://github.com/wwylab/DeMixT) 

Please let us know if there is anything else that we should do. 

Thanks,
Fan

**Comment by fgao-github** [comment]
Hello Bioconductor Team,

This is a follow-up email about our DeMixT package  
(https://github.com/wwylab/DeMixT). We use your standard data  
structure (SummarizedExperiment), and uploaded the vignette to the  
/vignettes directory on GitHub two weeks ago. Could you please reopen  
the issue for us? Thank you very much for your help. I look forward to  
hearing from you.

Thanks,
Fan


Quoting Fan Gao <fg15@rice.edu>:

> Hi Martin,
>
> Have you made any progress about our DeMixT package?  
> (https://github.com/wwylab/DeMixT)
>
> Please let us know if there is anything else that we should do.
>
> Thanks,
> Fan
>
>

**Comment by mtmorgan** [comment]
Please remember that your package should have version number 0.99.0, and after each commit where you would like to see a new build (so that you can check your changes), the version should be bumped, to 0.99.1, 0.99.2, 0.99.3, ...

**Comment by fgao-github** [comment]
Hi Martin,

Thank you for your quick reply. I just changed the version number to  
0.99.0. Could you help please check if there is anything else we need  
to do?

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
# 14 March, 2019

- Most comments below are 'examples', with code changes implied at several places

- Please update the DESCRIPTION file to correctly capitalize

    ```
    VignetteBuilder: knitr
    ```
    
- Please build and check your package using R-devel, e.g., 
    ```
    R CMD build DeMixT
    R CMD check DeMixT_0.99.0.tar.gz
    ```
  please be sure to address each issue flagged by R CMD check.
  
- Please install and run BiocCheck on your package

    ```
    > BiocManager::install("BiocCheck")
    
    $ R CMD BiocCheck DeMixT
    ```

- provide a Description: that reads more like a paper abstract, describing the unique features of your package

- demixt.Rmd:96 these data sets are not provided in your package

- as 'best practice', include the output of `sessionInfo()` at the end of your vignette.

- DeMixT.c:21 instead of printing results from C, return a value and use `packageStartupMessage()` to communicate with the user.

- DeMixT.R:1 do not use `require()` in your code; it is sufficient to include packages as Depends: or Imports:

- DeMixT.R:9 use `message()` (prints to stderr) rather than `cat()` (prints to stdout) for communicating with users

- DeMixT.R:22 avoid very long lines that are difficult to read

- DeMixT.S1.R:4 avoid naming conventions that look like S3 methods, e.g., use DeMixT_S1

- DeMixT.S1.R:101 use efficient matrixStats and base functions to avoid iteration, e.g., `rowMeans()`, `rowSds()`

- man pages: good.

**Comment by fgao-github** [comment]
Hi Martin,

I have modified our DeMixT package based on your comments. I just  
uploaded the modified package on GitHub  
(https://github.com/wwylab/DeMixT)

Could you please check and let us know if there are any questions?

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
Please commit a 'version bump' (change the Version: field in the DESCRIPTION file) to 0.99.1. This will trigger an automatic build. Before commiting the version bump, make sure the [web hook][1] is installed correctly.

[1]: https://github.com/Bioconductor/Contributions#adding-a-web-hook

**Comment by fgao-github** [comment]
Hi Martin,

We have installed the web hood and changed the version to 0.99.1.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
Did you update the webhook before the version bump? Can you verify that step 8 of https://github.com/Bioconductor/Contributions#adding-a-web-hook worked? The version bump did not trigger an automatic build. If you update the web hook, please increment the version (to 0.99.2) to again try to trigger an automatic build.

**Comment by fgao-github** [comment]
Hi Martin,

We have set up the step 8 of web hook. Please see attached. I also  
changed the version to 0.99.2.

Please help check if it works.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by fgao-github** [comment]
Hi Martin,

Our DeMixT package (0.99.9) has been built without errors or warnings.
(http://bioconductor.org/spb_reports/DeMixT_buildreport_20190402154834.html)

Please let me know if there is anything else we need to do.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/fgao-github.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("DeMixT")`. The package 'landing page' will be created at

https://bioconductor.org/packages/DeMixT

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/

**Comment by fgao-github** [comment]
Hi Martin,

I already set up ssh public keys for my GitHub account (fgao-github).

Please let me know if you have any questions.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by fgao-github** [comment]
Hi Martin,

I found that the information about our DeMixT package on your website  
does not reflect our recent changes on the GitHub.
https://bioconductor.org/packages/release/bioc/html/DeMixT.html
https://github.com/wwylab/DeMixT

Could you please help update, or let us know how to update the  
information on your website?

Thansk,
Fan


Quoting Fan Gao <fgao@rice.edu>:

**Comment by mtmorgan** [comment]
Are you remembering to push changes to git@git.bioconductor.org:packages/DeMixT, as you have done in the past?

**Comment by fgao-github** [comment]
Hi Martin,

We did not pushed changes before, and the information on your website  
remained the same since our package got approved.

I followed the instructions on this website:
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

I updated the content on the github:
https://github.com/wwylab/DeMixT/tree/RELEASE_3_9

I will check the website tomorrow after 6 pm EDT.
https://bioconductor.org/packages/release/bioc/html/DeMixT.html

Please let me know if there is anything else I need to do.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
You need to push changes to git.bioconductor.org, which is the canonical location for your package. In the instructions that you mention, did you add the 'upstream' repository (step 2), merge, update and then push (step 9) to both upstream and origin repositories?

**Comment by mtmorgan** [comment]
OK, I see now new commits on git.bioconductor.org.

Note also that I see commits for

 .DS_Store           | Bin 0 -> 6148 bytes
 DESCRIPTION~        |  20 ++++++
 inst/.DS_Store      | Bin 0 -> 6148 bytes

These files should not be in git, so please remove. Consider investigating and using a .gitignore file in the root directory of your package ensure that similar files are not committed in the future.

Since you did not increment the version of your package (to 1.1.1 and subsequently 1.1.2, 1.1.3, ...) on the master branch), the package will build on the nightly build report, but will not push to the public repository. Also, you made commits to the master branch of git.bioconductor.org. This is appropriate for introducing bug fixes and new features, and these appear in the devel branch of bioconductor, so you should be looking for correct builds and updated versions on the devel branch

  http://bioconductor.org/checkResults/devel/bioc-LATEST/DeMixT/
  http://bioconductor.org/packages/devel/bioc/html/DeMixT.html

If the RELEASE_3_9 branch has been updated with bug fixes (ONLY), then the version there should be updated to 1.0.1 (1.0.2, 1.0.3, ...); the updates will appear at the above URLs, but with devel replaced with 'release'.

**Comment by fgao-github** [comment]
Hi Martin,

We just wanted to update the citation info (citation("DeMixT")) on  
your website. Currently we updated it on the two branches in  
https://github.com/wwylab/DeMixT.

I just tried to increment the version number to push to your public  
repository. I got the following error message.
remote: Error: Illegal version bump from '1.0.0' to '1.1.1'.

Then, I changed the version number and push again. I got a similar message.
remote: Error: Illegal version bump from '1.1.1' to '1.0.1'.

I am very confused about which version number I should use. Should I  
use 1.2.0 and push again?

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
If you're pushing to the RELEASE_3_9 branch, then the version number should be 1.0.1. If you're pushing to the devel / master branch, the version should be 1.1.1. 

What does the DESCRIPTION file look like in the commit that you are trying to push? What branch are you trying to push to? It might be helpful to look at

  git branch -a

and 

  git remote -v

**Comment by fgao-github** [comment]
Hi Martin,

I did not see any changes in the following websites.
http://bioconductor.org/packages/release/bioc/html/DeMixT.html
http://bioconductor.org/packages/devel/bioc/html/DeMixT.html

I attached the screenshot of the git commands with this email.

I will try pushing the 1.0.1 to the upstream/RELEASE_3_9 tonight, and  
see if it works tomorrow.

Please correct me if I was wrong.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
Now that your package is accepted, it builds nightly. Check out http://bioconductor.org/checkResults/devel/bioc-LATEST/DeMixT/ where you can compare the date the page was generated to the current date, and you can see the 'snapshot' date when the nightly build started and compare that to the commits that you made. You can also look for a green 'light' to the very right, which indicates both that the package build and that it was pushed.

As of now, the build report was produced on Tuesday from builds that started on Monday at 17:00 (Eastern), before you made your commits. So you'll have to wait until Wednesday.

**Comment by fgao-github** [comment]
Hi Martin,

I encountered a problem when pushing a newer version of DeMixT (1.0.2)  
to bioconductor.

$git push upstream RELEASE_3_9

error: src refspec RELEASE_3_9 does not match any.
error: failed to push some refs to  
'git@git.bioconductor.org:packages/DeMixT.git'

Besides, I think I successfully pushed 1.1.2 to the level branch.  
Could you please check if you received it?

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
now that your package is accepted, please ask questions on the bioc-devel mailing list https://stat.ethz.ch/mailman/listinfo/bioc-devel

**Comment by fgao-github** [comment]
Hi Martin,

Could you please help us remove or let me know how to remove the  
vignette file in pdf on the website?
https://bioconductor.org/packages/release/bioc/html/DeMixT.html

The format (font size, figures, etc) is quite different between the  
pdf file generated on my machine and the one generated automatically  
from the website, using exactly the same RMD file. So I changed the  
format to html file, which looks better, but the outdated pdf file  
could not be removed.

I appreciate your help very much. Let me know if you need any  
additional information.

Thanks,
Fan


Quoting Martin Morgan <notifications@github.com>:

**Comment by mtmorgan** [comment]
Please see my previous comment about posting to the bioc-devel mailing list.

**Comment by pengyang0411** [comment]
Hello Bioconductor Team,

I am a new maintainer of DeMixT package.

We aimed to release a new updated DeMixT package from version 1.3.1 to 1.5.0 in devel branch.
However, I got the following error message when I try to push the package public:
Error: Illegal version bump from '1.3.1' to '1.5.0'.

I am very confused about this issue, since I noticed that we need use odd number for y(x.y.z) in devel. How can I address this issue?

Thanks,
Peng
